release_dashboard/lib/widgets/common/continue_button.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+typedef DynamicFuture = Future<dynamic> Function();
+
+/// The continue button of each step.
+///
+/// [elevatedButtonKey] is assigned to [ElevatedButton]. It makes locating and
+/// pressing the button easier in tests.
+///
+/// [error] is the error string that gets displayed when there is an error in the release.
+/// procedure. The error text is in red and displayed above [ElevatedButton].
+///
+/// When [enabled] is true, [ElevatedButton] is enabled, disabled otherwise.
+///
+/// [onPressedCallback] will be executed when [ElevatedButton] is enabled and pressed.
+/// [onPressedCallback] must be asynchronous.
+///
+/// when [isLoading] is true, [CircularProgressIndicator] will be dispalyed, hidden otherwise.
+class ContinueButton extends StatelessWidget {
+  const ContinueButton({
+    Key? key,
+    required this.elevatedButtonKey,
+    this.error,
+    required this.enabled,
+    required this.onPressedCallback,
+    required this.isLoading,
+  }) : super(key: key);
+
+  final Key elevatedButtonKey;
+  final String? error;
+  final bool enabled;
+  final DynamicFuture onPressedCallback;
+  final bool isLoading;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        if (error != null)
+          Center(
+            child: SelectableText(
+              error!,
+              style: Theme.of(context).textTheme.subtitle1!.copyWith(color: Colors.red),
+            ),
+          ),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            ElevatedButton(
+              key: elevatedButtonKey,
+              onPressed: enabled == true
```

Comment:
When `isLoading` is true, what marks it back to false? Usually you would confirm the `DynamicFuture` is completed.

Reply:
The parent can change it back. Because it is passed as the parent's state. For example here: 
https://github.com/Yugue/cocoon/blob/eaed2969a2accc1cbf772d1eb1cb73bccda84599/release_dashboard/lib/widgets/create_release_substeps.dart#L221

Reply:
The parent shouldn't have control over this, or need to code it. You can use `Future.whenComplete` to mark it as finished.

https://api.dart.dev/stable/2.14.4/dart-async/Future-class.html

Reply:
The parent needs to control the enabling behaviour of the continue button. For `CreateReleaesSubsteps`, when all the inputs and dropdowns of the parent are validated, the continue button becomes enabled. For other steps, the continue button enables when all the parent's checkboxes are checked.